config/v1alpha2/types_insights.go

```diff
@@ -0,0 +1,194 @@
+	// The PersistentVolumeClaim must be created in the openshift-insights namespace.
+	// +required
+	Claim PersistentVolumeClaimReference `json:"claim"`
+	// mountPath is an optional field specifying the directory where the PVC will be mounted inside the Insights data gathering Pod.
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
+	// The current default mount path is /var/lib/insights-operator
+	// The path may not exceed 1024 characters and must not contain a colon.
+	// +kubebuilder:validation:MaxLength=1024
+	// +kubebuilder:validation:XValidation:rule="!self.contains(':')",message="mountPath must not contain a colon"
+	// +optional
+	MountPath string `json:"mountPath,omitempty"`
+}
+
+// persistentVolumeClaimReference is a reference to a PersistentVolumeClaim.
+type PersistentVolumeClaimReference struct {
+	// name is a string that follows the DNS1123 subdomain format.
+	// It must be at most 253 characters in length, and must consist only of lower case alphanumeric characters, '-' and '.', and must start and end with an alphanumeric character.
+	// +kubebuilder:validation:XValidation:rule="!format.dns1123Subdomain().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	// +kubebuilder:validation:MaxLength:=253
+	// +required
+	Name string `json:"name"`
+}
+
+// gathererConfig allows to configure specific gatherers
+type GathererConfig struct {
+	// name is the required name of a specific gatherer
+	// It may not exceed 256 characters.
+	// The format for the disabledGatherer should be: {gatherer}/{function} where the function is optional.
+	// Gatherer consists of a lowercase letters only that may include underscores (_).
+	// Function consists of a lowercase letters only that may include underscores (_) and is separated from the gatherer by a forward slash (/).
+	// The particular gatherers can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
+	// Run the following command to get the names of last active gatherers:
+	// "oc get insightsoperators.operator.openshift.io cluster -o json | jq '.status.gatherStatus.gatherers[].name'"
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:XValidation:rule=`self.matches("^[a-z]+[_a-z]*[a-z]([/a-z][_a-z]*)?[a-z]$")`,message=`gatherer name must be in the format of {gatherer}/{function} where the gatherer and function are lowercase letters only that may include underscores (_) and are separated by a forward slash (/) if the function is provided`
+	// +required
+	Name string `json:"name"`
+	// state is an optional field that allows you to configure specific gatherer. Valid values are "Enabled" and "Disabled".
```

Comment:
From an earlier comment by Joel, it sounds like this API has been around for a little while. When we can I think we try to avoid `Enabled` and `Disabled` as they can be overloaded terms. I think it _can_ make sense in this case and the GoDoc describes what each state represents.

That being said, is there any other configuration option that could be more descriptive than `Enabled`/`Disabled`?

Reply:
I think not at the moment. We simply want to be able to disable some of the data points/gatherers.

Reply:
If you just want to be able to disable gatherers, and the default is that all are enabled, does it make more sense to just have a slice of gatherer names that should be disabled?

For example:
```yaml
disabledGatherers:
  - foo/bar
  - baz
  ...
```
instead of:
```yaml
gatherers:
  - name: foo/bar
    state: Disabled
  - name: baz
    state: Disabled
  ...
```

Reply:
The map based approach was introduced to allow additional configuration for gatherers to be included in this list in the future.

If we decide in the future to add additional gather config for specific gatherers, and have a disabled list, then there's a disconnect between where the gatherer is enabled/disabled and the configuration of the gatherer itself

With the current pattern we can do something like below, where we disable config from being set when state is Disabled
```
gatherers:
- name: foo
  state: Enabled
  config:
    gatherer: foo
    foo:
      ... # specific options
```

Reply:
I will stick with the map-based approach here to be more flexible for future configuration.